app/controllers/assessments_controller.rb

```diff
@@ -5,83 +5,85 @@ class AssessmentsController < ApplicationController
   def index; end
 
   def new
-    # Don't bother with this if the submission token isn't the correct length
-    @patient_submission_token = params[:patient_submission_token].gsub(/[^0-9a-z]/i, '')
-    return if @patient_submission_token.length != 40
+    # Validate and get patient submission token and redirect if invalid link or already reported
+    @patient_submission_token = check_and_get_patient_submission_token(invalid_link_url,
+                                                                       ADMIN_OPTIONS['report_mode'] ? already_reported_report_url : already_reported_url)
+    return if @patient_submission_token.nil?
 
     # Don't bother with this if the jurisdiction unique identifier isn't at least 10 characters long
-    @unique_identifier = params[:unique_identifier]&.gsub(/[^0-9a-z]/i, '')
-    return if @unique_identifier.present? && @unique_identifier.length < 10
+    @unique_identifier = params.permit(:unique_identifier)[:unique_identifier]&.gsub(/[^0-9A-Za-z_-]/i, '')
+    redirect_to(invalid_link_url) && return if @unique_identifier.present? && @unique_identifier.length < 10
 
-    @assessment = Assessment.new
-
-    # If monitoree, limit number of reports per time period
-    if current_user.nil? && AssessmentReceipt.where(submission_token: @patient_submission_token)
-                                             .where('created_at >= ?', ADMIN_OPTIONS['reporting_limit'].minutes.ago)
-                                             .exists?
-
-      redirect_to(already_reported_report_url) && return if ADMIN_OPTIONS['report_mode']
+    # Replace old unique identifier with new unique identifier if applicable
+    if @unique_identifier.present? && @unique_identifier.length > 10
+      jurisdiction_lookup = JurisdictionLookup.where('old_unique_identifier like ?', "#{@unique_identifier}%").first
```

Comment:
I believe it's because the current unique_identifier is actually even longer but the version of it sent in the link is just a substring (first 32 characters) of it. @mayerm94 correct me if I'm wrong